### DISCUSSION TITLE
how to revalidate data in server component?

### DISCUSSION THREAD

**Discussion by nicitaacom** [vercel/next.js#58812] [Help] [answered]
### Summary

Next js question

I have a ticketId which I pass to the client component in layout.tsx

I have a route handler to close a ticket
```ts
export async function POST(req: Request) {
   const { ticketId, rate } = (await req.json()) as TAPITicketsRate
   const { error } = await supabaseAdmin.from("tickets").update({ is_open: false, rate: rate }).eq("id", ticketId)
   if (error) return NextResponse.json({ error: `Error in api/tickets/route.ts\n ${error.message}` }, { status: 400 })
   await pusherServer.trigger("tickets", "tickets:close", {
     id: ticketId,
     is_open: false,
   } as ITicket)
   // TODO - after closing ticket don't show messages from currently closed ticket
   await getTicketId()
   revalidatePath("/")
   revalidateTag("layout.tsx")

   return NextResponse.json({ message: "Ticket marked as completed (ticket closed)" }, { status: 200 })
}
```

And the question is how can I not show in this client component where I pass the ticketId of messages that come from the server and are transmitted to this client component based on this ticketId

layout.tsx
```tsx
export default async function RootLayout({ children }: { children: React.ReactNode }) {
   const ticketId = await getTicketId()
   const initial_messages = await getInitialMessagesByTicketId(ticketId)

   return (
     <html lang="en">
       <body>
         <ClientOnly>
           <Layout>{children}</Layout>
           <SupportButton initialMessages={initial_messages ?? []} ticketId={ticketId} />
         </ClientOnly>
       </body>
     </html>
   )
}
```

In short, I want these lines
```ts

  const ticketId = await getTicketId()
   const initial_messages = await getInitialMessagesByTicketId(ticketId)
```

executed again when I do the logic for closing the ticket in the route handler
(executed again to pass initialMessages and a new ticketId to the client component

**Comment by icyJoseph** [comment, marked as answer]
I am not sure I follow your logic, but within the SupportButton, you probably have an on click handler, there you could `await` the call to the POST endpoint, and after that do a router.refresh(), or revalidate path call, that the client sees. The way you have it, you are telling the server to revalidate in preparation for future requests.

**Reply by nicitaacom** [reply]
That's work
Thank you

Have a great day

I'm still suprise how easy is that
I just added router.refresh() and trigger for pusher to `setMessages([])`
